core/state/statedb.go

```diff
@@ -450,20 +450,22 @@ func (self *StateDB) Copy() *StateDB {
 	self.lock.Lock()
 	defer self.lock.Unlock()
 
+	dirtyObjects := self.journal.flatten()
```

Comment:
This seems dangerous to me because it iterates over all the previous opcodes. So we have an O(opcodes * calls) complexity here

Reply:
Not 'opcodes' technically, just the ones that modify the state in some manner. Why would it be O( n*m) ? 
Afict it's O( n ) , n being the length of the journal.

Reply:
But yes, I'm not 100% certain about how to view `Copy`, and what the semantics of that should be. The copy should (?) have a clean and fresh journal, but needs a map of uncommitted changes.